pkg/pool-manager/pool.go

```diff
@@ -176,3 +192,42 @@ func getNextMac(currentMac net.HardwareAddr) net.HardwareAddr {
 
 	return currentMac
 }
+
+// Checks if the namespace of an instance is opted in for kubemacpool
+func (p *PoolManager) isInstanceOptedIn(namespaceName, labelSelector, labelValue string) (bool, error) {
+	log.V(3).Info("checking for label in namespace", "webhook labelSelector", labelSelector, "webhook labelValue", labelValue)
+	isNamespaceOptedIn := false
+
+	ns, err := p.kubeClient.CoreV1().Namespaces().Get(namespaceName, metav1.GetOptions{})
+	if err != nil {
+		return isNamespaceOptedIn, err
```

Comment:
Add context to errors with `errors.Wrap` so it's easier to debug and see where the error comes from.